### ISSUE TITLE
Investigate: Puma Rack tests failing on 3.2.0-preview2

### ISSUE THREAD

**Issue by tannalynn** [newrelic/newrelic-ruby-agent#1446]
On the new 3.2.0-preview2 version, all rack test suites involving puma are [failing](https://github.com/newrelic/newrelic-ruby-agent/actions/runs/3066671041/jobs/4952134953) with the following errors:
```
  1) Error:
RackAutoInstrumentationTest#test_middlewares_record_metrics:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/test/multiverse/suites/rack/example_app.rb:84:in `initialize'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/middleware_proxy.rb:32:in `new'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/middleware_proxy.rb:32:in `new'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:86:in `block in use'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:147:in `block in to_app'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:147:in `each'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:147:in `inject'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:147:in `to_app'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/rack/prepend.rb:24:in `block in to_app'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/rack/instrumentation.rb:44:in `with_deferred_dependency_detection'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/lib/new_relic/agent/instrumentation/rack/prepend.rb:24:in `to_app'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-1.6.13/lib/rack/builder.rb:59:in `app'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/test/multiverse/suites/rack/rack_auto_instrumentation_test.rb:28:in `app'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-test-2.0.2/lib/rack/test/methods.rb:45:in `build_rack_test_session'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-test-2.0.2/lib/rack/test/methods.rb:33:in `rack_test_session'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/gems/3.2.0+2/gems/rack-test-2.0.2/lib/rack/test/methods.rb:52:in `current_session'
    /opt/hostedtoolcache/Ruby/3.2.0-preview2/x64/lib/ruby/3.2.0+2/forwardable.rb:234:in `get'
    /home/runner/work/newrelic-ruby-agent/newrelic-ruby-agent/test/multiverse/suites/rack/rack_auto_instrumentation_test.rb:91:in `test_middlewares_record_metrics'

  2) Error:
RackAutoInstrumentationTest#test_body_is_preserved:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  3) Error:
RackAutoInstrumentationTest#test_non_agent_middlewares_do_not_record_metrics_if_disabled_by_config:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  4) Error:
RackAutoInstrumentationTest#test_middlewares_record_queue_time:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
...

  5) Error:
RackAutoInstrumentationTest#test_middleware_gets_used:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  6) Error:
RackAutoInstrumentationTest#test_header_is_preserved:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  7) Error:
RackAutoInstrumentationTest#test_middleware_that_returns_early_records_middleware_rollup_metric:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  8) Error:
RackAutoInstrumentationTest#test_middleware_created_with_args_works:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....

  9) Error:
RackAutoInstrumentationTest#test_status_code_is_preserved:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
...

 10) Error:
RackAutoInstrumentationTest#test_middleware_that_returns_early_middleware_all_has_correct_call_times:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
...

 11) Error:
RackAutoInstrumentationTest#test_middleware_created_with_kwargs_works:
ArgumentError: wrong number of arguments (given 2, expected 1; required keyword: tag)
....
```

**Comment by fallwith** [comment]
3.2 went stable a few months after this issue was created and we've been sailing smoothly with it being used for our Puma CI tests for quite awhile now. Closing as resolved.

There is another issue this year with the 3.3 preview release. That issue was traced to Puma's fork of Rack, which we have addressed with a PR (merged now) submitted to the Puma repo.

For 3.3, we're now simply waiting on a new Puma release to re-enable 3.3 based tests. That waiting is being tracked with https://github.com/newrelic/newrelic-ruby-agent/issues/2042
